src/ReservoirPriceOracle.sol

```diff
@@ -1,32 +1,405 @@
+    error RPC_INVALID_ROUTE_LENGTH();
+    error RPC_INVALID_ROUTE();
+    error RPC_UPDATE_PRICE_NOT_SIMPLE_ROUTE();
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                                        STORAGE                                            //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    /// @notice percentage change greater than which, a price update with the oracles would succeed
+    /// 1e18 == 100%
+    uint64 public priceDeviationThreshold;
+
+    /// @notice multiples of the base fee the contract rewards the caller for updating the price when it goes
+    /// beyond the `priceDeviationThreshold`
+    uint64 public rewardMultiplier;
+
+    /// @notice TWAP period for querying the oracle in seconds
+    uint64 public twapPeriod;
+
+    /// @notice Designated pairs to serve as price feed for a certain token0 and token1
+    mapping(address token0 => mapping(address token1 => ReservoirPair pair)) public pairs;
+
+    /// @notice The latest cached geometric TWAP of token1/token0, where the address of token0 is strictly less than the address of token1
+    /// Stored in the form of a 18 decimal fixed point number.
+    /// Supported price range: 1wei to 1e36, due to the need to support inverting price via `Utils.invertWad`
+    /// To obtain the price for token0/token1, calculate the reciprocal using Utils.invertWad()
+    mapping(address token0 => mapping(address token1 => uint256 price)) public priceCache;
+
+    /// @notice Defines the route to determine price of token0, where the address of token0 is strictly less than the address of token1
+    mapping(address token0 => mapping(address token1 => address[] path)) private _route;
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                                CONSTRUCTOR, FALLBACKS                                     //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    constructor(uint64 aThreshold, uint64 aTwapPeriod, uint64 aMultiplier) {
+        updatePriceDeviationThreshold(aThreshold);
+        updateTwapPeriod(aTwapPeriod);
+        updateRewardMultiplier(aMultiplier);
+    }
+
+    /// @dev contract will hold native tokens to be distributed as gas bounty for updating the prices
+    /// anyone can contribute native tokens to this contract
+    receive() external payable { }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                                       MODIFIERS                                           //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    modifier validateTokens(address aTokenA, address aTokenB) {
+        if (aTokenA == aTokenB) revert RPC_SAME_TOKEN();
+        _;
+    }
```

Comment:
What's the rationale here?

Reply:
the original intention was for error checking. But now it seems that in all the instances where it is used, the functions check for the existence of a path anyway. So if they were the same the functions would revert with `NoPath()` anyway. 

Deleting this modifier.